bolt/lib/Passes/LongJmp.cpp

```diff
@@ -382,6 +385,11 @@ uint64_t LongJmpPass::tentativeLayoutRelocMode(
     DotAddress += Func->estimateConstantIslandSize();
     ++CurrentIndex;
   }
+  if (!ColdLayoutDone) {
```

Comment:
Instead of the if() on line 360 gating whether to skip the non-emitted function, if the goal is really to ensure that the layout for the cold part of split functions is done, why not complete the cold layout here on line 388:

```
  if (!ColdLayoutDone) {
    DotAddress =
          tentativeLayoutRelocColdPart(BC, SortedFunctions, DotAddress);
    if (opts::HotFunctionsAtEnd)
        DotAddress = alignTo(DotAddress, opts::AlignText);
  }
```

Reply:
I see your point.

What I had in mind when writing the patch was to introduce a minimal change in the logic.
I'm not sure if running `tentativeLayoutRelocColdPart` at 388 would require any of the remaining code between 367-385 to run as well.

I would want some feedback from @yota9  on this, to see how can we refactor/simplify the pass he originally authored.

Reply:
Hello. I would take a look into it. But the pass belongs to @rafaelauler , I only made a few bugfixes here :)

Reply:
Thanks @yota9 .

BTW, I've just released a reproducer on the bug I describe earlier on this PR (see [point (2)](https://github.com/llvm/llvm-project/pull/96609#issuecomment-2188867516) in my initial comment)

**The issue is:**
- #99848

A possible lead could be the FixRelaxations pass. Any feedback would be appreciated (cc: @yota9 @rafaelauler  @samolisov ).

Reply:
After checking out the review again I agree with @rafaelauler. I would rather create lamda with 
`
      DotAddress =
          tentativeLayoutRelocColdPart(BC, SortedFunctions, DotAddress);
      ColdLayoutDone = true;
      if (opts::HotFunctionsAtEnd)
        DotAddress = alignTo(DotAddress, opts::AlignText);
`

And run it under previous condition and under only `if (!ColdLayoutDone)`  after the loop.

Reply:
Done, thanks!